sympy/combinatorics/homomorphisms.py

```diff
@@ -417,3 +420,128 @@ def block_homomorphism(group, blocks):
+    >>> from sympy.combinatorics.free_groups import free_group
+    >>> from sympy.combinatorics.fp_groups import FpGroup
+    >>> from sympy.combinatorics.homomorphisms import homomorphism, group_isomorphism
+    >>> from sympy.combinatorics.named_groups import DihedralGroup, AlternatingGroup
+
+    >>> D = DihedralGroup(8)
+    >>> p = Permutation(0, 1, 2, 3, 4, 5, 6, 7)
+    >>> P = PermutationGroup(p)
+    >>> group_isomorphism(D, P)
+    (False, None)
+
+    >>> F, a, b = free_group("a, b")
+    >>> G = FpGroup(F, [a**3, b**3, (a*b)**2])
+    >>> H = AlternatingGroup(4)
+    >>> (check, T) = group_isomorphism(G, H)
+    >>> check
+    True
+    >>> T(b*a*b**-1*a**-1*b**-1)
+    (0 2 3)
+
+    '''
+    if not isinstance(G, (PermutationGroup, FpGroup)):
+        raise TypeError("The group must be a PermutationGroup or a finite FpGroup")
+    if not isinstance(H, (PermutationGroup, FpGroup)):
+        raise TypeError("The group must be a PermutationGroup or a finite FpGroup")
+
+    if isinstance(G, FpGroup) and isinstance(H, FpGroup):
+        G = simplify_presentation(G)
+        H = simplify_presentation(H)
+        # Two infinite FpGroups with the same generators are isomorphic
+        # when the relators are same but are ordered differently.
+        if G.generators == H.generators and (G.relators).sort() == (H.relators).sort():
+            if not isomorphism:
+                return True
+            return (True, homomorphism(G, H, G.generators, H.generators))
+
+    #  `_H` is the permutation groups isomorphic to `H`.
+    _H = H
+    h_isomorphism = None
+    g_order = G.order()
+    h_order = H.order()
+
+    if g_order == S.Infinity:
+        raise NotImplementedError("Isomorphism methods are not implemented for infinite groups.")
+
+    if isinstance(H, FpGroup):
+        if h_order == S.Infinity:
+            raise NotImplementedError("Isomorphism methods are not implemented for infinite groups.")
+        _H, h_isomorphism = H._to_perm_group()
+
+    if (g_order != h_order) and (G.is_abelian != H.is_abelian):
+        if not isomorphism:
+            return False
+        return (False, None)
+
+    if not isomorphism:
+        # Two groups of same cyclic numbered order
+        # are ismorphic to each other.
+        n = g_order
+        if (g_order == h_order) and (igcd(n, totient(n))) == 1:
+            return True
+
+    # Match the generators of `_G` with the subsets of `_H`
+    gens = G.generators
+    for subset in itertools.permutations(_H, len(gens)):
+        gens = list(gens)
+        images = list(subset)
+        images.extend([_H.identity]*(len(G.generators)-len(images)))
+        gens.extend([g for g in G.generators if g not in gens])
+        _images = dict(zip(gens,images))
+        if _check_homomorphism(G, _H, _images):
+            images = []
+            if isinstance(H, FpGroup):
+                images = h_isomorphism.invert(list(subset))
+            else:
+                images = subset
```

Comment:
You don't need this line at all if you don't rewrite `images` with `[]` as you do before the `if`.